### PR TITLE
Realtime Activity Updates

### PR DESCRIPTION
**Pull request by ananyeet**
### What is a quick description of the change?

Update the activity viewer to update upon change to the database. 

### Is this fixing an issue?

none.

### Are there more details that are relevant?

This was manually tested by two people making updates to the same activity and watching it change in realtime. 

### Check lists (check `x` in `[ ]` of list items)

- [x] Test written/updating
- [x] Tests passing
- [x] Coding style (indentation, etc)

No additional tests were written with these changes.

### CODE REVIEW

File: frontend/src/components/ViewActivities/activitylist.js [hunk 1]
```diff
@@ -8,17 +8,32 @@ import app from '../Firebase';
 const db = app.firestore();
 
 /**
- * Gets the list of activities from the server. 
+ * The list of activities. 
  * 
- * @param {string} tripId The trip ID.
+ * @param {Object} props This component expects the following props:
```
[the remark this answers is not in the document]
Comment (ananyeet) [reply frontend/src/components/ViewActivities/activitylist.js:13]: ooooh I forgot about that!!

File: frontend/src/components/ViewActivities/activitylist.js [hunk 2]
```diff
@@ -33,25 +48,10 @@ export async function getActivityList(tripId) {
...
- * - `tripId` {string} The trip's ID.  
- */
-class ActivityList extends React.Component {
-  /** @inheritdoc */
-  constructor(props) {
-    super(props);
-    this.state = { days : [] };
+      });
+      this.setState({ days: activityFns.sortByDate(tripActivities) });
+    });
```
Comment (zghera) [inline frontend/src/components/ViewActivities/activitylist.js:53]: If you want, you can pass your error callback into the second parameter of onSnapshot [like so](https://firebase.google.com/docs/firestore/query-data/listen#handle_listen_errors).

File: frontend/src/components/ViewActivities/activitylist.js [hunk 3]
```diff
@@ -33,25 +48,10 @@ export async function getActivityList(tripId) {
...
- * @param {Object} props This component expects the following props:
- * - `tripId` {string} The trip's ID.  
- */
-class ActivityList extends React.Component {
-  /** @inheritdoc */
-  constructor(props) {
-    super(props);
-    this.state = { days : [] };
+      });
+      this.setState({ days: activityFns.sortByDate(tripActivities) });
```
Comment (zghera) [inline frontend/src/components/ViewActivities/activitylist.js:52]: Nice job debugging this!

File: frontend/src/components/ViewActivities/activitylist.js [hunk 4]
```diff
@@ -8,17 +8,32 @@ import app from '../Firebase';
...
+  /** @inheritdoc */
+  constructor(props) {
+    super(props);
+    this.state = { days : [] };
+
+    this.getActivityList = this.getActivityList.bind(this);
+  }
+
+  /**
+   * Gets the list of activities from the server.
```
Comment (zghera) [inline frontend/src/components/ViewActivities/activitylist.js:26]: You may want to mention what state you set in here. That is up to you though so don't have to add if you don't think it is important.

File: frontend/src/components/ViewActivities/activitylist.js (hunk 2 above)
Comment (ananyeet) [reply frontend/src/components/ViewActivities/activitylist.js:53]: Yeah. Error behavior is specified in render() though.

File: frontend/src/components/ViewActivities/activitylist.js [hunk 5]
```diff
@@ -8,17 +8,33 @@ import app from '../Firebase';
...
+   * This function sets `this.state.days` to the sorted days.
+   * 
+   * @param {string} tripId The trip ID.
+   */
+  async getActivityList(tripId) {
+    db.collection(DB.COLLECTION_TRIPS).doc(tripId)
+    .collection(DB.COLLECTION_ACTIVITIES)
+    .onSnapshot(querySnapshot => {
+      let tripActivities = [];
+      this.setState({days: []});
```
Comment (Hiram-Silvey) [inline frontend/src/components/ViewActivities/activitylist.js:37]: What does this line do?

File: frontend/src/components/ViewActivities/activitylist.js [hunk 6]
```diff
@@ -27,29 +43,18 @@ export async function getActivityList(tripId) {
...
         // TODO: if start date != end date, split into 2 days. (#37)
 
-      // Eliminate nanoseconds, convert to milliseconds.
-      data[DB.ACTIVITIES_START_TIME] =
-        data[DB.ACTIVITIES_START_TIME]['seconds'] * 1000;         
-      data[DB.ACTIVITIES_END_TIME] = 
-        data[DB.ACTIVITIES_END_TIME]['seconds'] * 1000;
-
-      tripActivities.push(data);
+        if (data[DB.ACTIVITIES_START_TIME] !== undefined) {// not in new mode
```
Comment (Hiram-Silvey) [inline frontend/src/components/ViewActivities/activitylist.js:46]: Add space after `{`, capitalize `n` in `not` and add period.

File: frontend/src/components/ViewActivities/activitylist.js (hunk 5 above)
Comment (ananyeet) [reply frontend/src/components/ViewActivities/activitylist.js:37]: Clear the current state of the activities. Prevents everything from showing up twice every time there's an edit.

**Pull request merged**